q2_amr/card/partition.py

```diff
@@ -0,0 +1,205 @@
+import os
+import warnings
+from typing import Union
+
+import numpy as np
+from qiime2.util import duplicate
+
+from q2_amr.types import (
+    CARDAlleleAnnotationDirectoryFormat,
+    CARDAnnotationDirectoryFormat,
+    CARDGeneAnnotationDirectoryFormat,
+    CARDMAGsKmerAnalysisDirectoryFormat,
+    CARDReadsAlleleKmerAnalysisDirectoryFormat,
+    CARDReadsGeneKmerAnalysisDirectoryFormat,
+)
+
+
+def partition_mags_annotations(
+    annotations: CARDAnnotationDirectoryFormat, num_partitions: int = None
+) -> CARDAnnotationDirectoryFormat:
+    return _partition_annotations(annotations, num_partitions)
+
+
+def partition_reads_allele_annotations(
+    annotations: CARDAlleleAnnotationDirectoryFormat, num_partitions: int = None
+) -> CARDAlleleAnnotationDirectoryFormat:
+    return _partition_annotations(annotations, num_partitions)
+
+
+def partition_reads_gene_annotations(
+    annotations: CARDGeneAnnotationDirectoryFormat, num_partitions: int = None
+) -> CARDGeneAnnotationDirectoryFormat:
+    return _partition_annotations(annotations, num_partitions)
+
+
+def _partition_annotations(
```

Comment:
Do I see it correctly that your if-statements always have two branches: one for `CARDAnnotationDirectoryFormat` and another one for everything else? If that's the case, I think this whole function should really be broken down a bit - it's really long and could use some refactor and so that would be one way to think about refactoring it.